### DISCUSSION TITLE
Select with order by

### DISCUSSION THREAD

**Discussion by engesolda** [maxtoroq/DbExtensions#8] [Q&A] [answered]
Hi, I have this sql:
Ps.: The rowNumber value are just for example.

SELECT \* FROM(
SELECT ROW_NUMBER() OVER (ORDER BY id desc) AS rowNumber, \* 
FROM Perfil) as p
WHERE rowNumber >= 4 AND rowNumber < 7

All right, in my code im pagging and sorting a gridview and I dont know wich collum will be clicked to order. So I created this:

query.SELECT("*")
        .FROM(SQL.SELECT("ROW_NUMBER() OVER (ORDER BY {0} desc) AS rowNumber, *", _orderBy).FROM("Perfil"), "p")
         .WHERE()
         ._("rowNumber >= {0} AND rowNumber < {1}", startRowIndex, lastRow)
         ._If(!string.IsNullOrEmpty(_descricao), "descricao LIKE {0}", "%" + _descricao + "%");

But that doesnt work at all. My return arent orderly and I dont know why.

**Comment by maxtoroq** [comment, marked as answer]
Placeholders are for parameters or subqueries, do a simple concat:

``` csharp
SQL.SELECT("ROW_NUMBER() OVER (ORDER BY " + orderBy + " desc) AS rowNumber, *").FROM("Perfil")
```

I suggest you validate `orderBy` to avoid SQL injection.

**Comment by engesolda** [comment]
Thanks man that works.
